blobs.go

```diff
@@ -0,0 +1,64 @@
+package distribution
+
+import (
+	"io"
+	"time"
+
+	"github.com/docker/distribution/digest"
+)
+
+// Blob describes a readable and seekable binary object. Most registry
+// objects, including layers and manifests can be accessible by digest in a
+// BlobService.
+type Blob interface {
+	// NOTE(stevvooe): http.ServeContent requires an efficient implementation
+	// of ReadSeeker.Seek(0, os.SEEK_END).
+	io.ReadSeeker
+	io.ReaderAt
+	io.WriterTo
+	io.Closer
+
+	// Descriptor identifies the layer.
+	Descriptor() Descriptor
+}
+
+// BlobWriter provides a handle for working with a blob store. Instances
+// should be obtained from BlobService.Writer and BlobService.Resume. If
+// supported by the store, a writer can be recovered with the uuid.
+type BlobWriter interface {
+	io.WriteSeeker
+	io.WriterAt
+	io.ReaderFrom
+	io.Closer
+
+	// UUID returns the identifier for this writer.
+	UUID() string
```

Comment:
It's not enforced, in that the type is a string. We can call this `ID`, if that makes it more generalized.